59.md

```diff
@@ -0,0 +1,253 @@
+for a given event, perform collaborative signing, and more.
+
+This NIP relies on [NIP-44](./44.md)'s versioned encryption algorithms.
+
+# Overview
+
+This protocol uses three main concepts to protect the transmission of a target event: `rumor`s, `seal`s, and `gift wrap`s.
+
+- A `rumor` is a regular nostr event, but is **not signed**. This means that if it is leaked, it cannot be verified.
+- A `rumor` is serialized to JSON, encrypted, and placed in the `content` field of a `seal`. The `seal` is then signed by the author of the note. The only information publicly available on a `seal` is who signed it, but not what was said.
+- A `seal` is serialized to JSON, encrypted, and placed in the `content` field of a `gift wrap`.
+
+This allows the isolation of concerns across layers:
+
+- A rumor carries the content but is unsigned, which means if leaked it will be rejected by relays and clients,
+  and can't be authenticated. This provides a measure of deniability.
+- A seal identifies the author without revealing the content or the recipient.
+- A gift wrap can add metadata (recipient, tags, a different author) without revealing the true author.
+
+# Protocol Description
+
+## 1. The Rumor Event Kind
+
+A `rumor` is the same thing as an unsigned event. Any event kind can be made a `rumor` by removing the signature.
+
+## 2. The Seal Event Kind
+
+A `seal` is a `kind:13` event that wraps a `rumor` with the sender's regular key. The `seal` is **always** encrypted
+to a receiver's pubkey but there is no `p` tag pointing to the receiver. There is no way to know who the rumor is for
+without the receiver's or the sender's private key. The only public information in this event is who is signing it.
+
+```js
+{
+  "id": "<id>",
+  "pubkey": "<real author's pubkey>",
+  "content": "<encrypted rumor>",
+  "kind": 13,
+  "created_at": 1686840217,
+  "tags": [],
+  "sig": "<real author's pubkey signature>"
+}
+```
+
+Tags MUST must always be empty in a `kind:13`. The inner event MUST always be unsigned.
+
+## 3. Gift Wrap Event Kind
+
+A `gift wrap` event is a `kind:1059` event that wraps any other event. `tags` MUST include a single `p` tag containing the recipient's public key.
+
+The goal is to hide the sender's information, the metadata, and the content of the original event from the public. The only public information is the receiver's public key.
+
+```js
+{
+  "id": "<id>",
+  "pubkey": "<random, one-time-use pubkey>",
+  "content": "<encrypted kind 13>",
+  "kind": 1059,
+  "created_at": 1686840217,
+  "tags": [["p", "<Receiver>"]],
+  "sig": "<random, one-time-use pubkey signature>"
+}
+```
+
+# Encrypting Payloads
+
+Encryption is done following NIP-44 on the JSON-encoded event. Place the the encryption payload in the `.content`
+of the wrapper event (either a `seal` or a `gift wrap`).
+
+# Other Considerations
+
+Because ephemeral keys are often used to sign `gift wrap`s, a `seal` MAY include a `wrap_privkey` tag that
```

Comment:
This `wrap_privkey` links sender and receiver in a verifiable manner and can be revealed by the receiver at any time without any damage to the receiver's data. Adding it requires trusting the receiver and thus, I believe it requires some privacy risk warning.

Reply:
I don't understand, if the key is ephemeral what information is being shared?

Reply:
If the key to the gift wrap is added to the seal, you are cryptographically linking them. The point of separating them was to unlink it. :)

Reply:
I see what you mean. Would putting it on the rumor be any better?

Reply:
I think the best way would be to do https://github.com/nostr-protocol/nips/pull/539 with the receiver's key. Or simply make it so any GiftWrap can be deleted by the `p`-tag.

Reply:
Neither of those works for groups, since the shared key is the recipient (neither does this I guess). So anyone in the group would be able to delete the message. In that scenario the actual publisher would also have to hold on to the generated key in order to delete the event. But I guess if clients implement deletes, it would be enough to delete the rumor's id instead of the wrap's.

Reply:
I've removed the paragraph, so resolved, but still unsolved.